code/modules/games/boardgame.dm

```diff
@@ -239,6 +256,23 @@ obj/item/weapon/board/attackby(obj/item/I as obj, mob/user as mob)
 /obj/item/weapon/checker/king/red
 	piece_color ="red"
 
+/obj/item/weapon/checker/attackby(obj/item/I as obj, mob/user as mob)
+	if(istype(I, /obj/item/weapon/storage/box/chess))
+		var/obj/item/weapon/storage/box/chess/b = I
+		var/turf/T = get_turf(src)
+		if(T)
+			for(var/obj/item/weapon/checker/c in T.contents)
+				if(istype(I, /obj/item/weapon/storage/box/chess/red))
+					if(c.piece_color == "red")
+						c.forceMove(b)
+					else
+						continue
+				else if(c.piece_color != "red")
+					c.forceMove(b)
+			to_chat(user, "<span class='notice'>You put all checker pieces into [b].</span>")
```

Comment:
put a \the before the [b]

Reply:
You don't need `\the` if you use an object. It's automatically preceded by an article if the object's name is set up properly. Which it usually is.